### PR TITLE
Added search and install packs to build process

### PR DESCRIPTION
**Pull request by dantavori**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
Added "search and install packs and their dependencies" functionality into the build process, without context to the rest of the flow.

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

### CODE REVIEW

File: Tests/configure_and_test_integration_instances.py [hunk 1]
```diff
@@ -649,6 +655,9 @@ def main():
     server_numeric_version = get_server_numeric_version(ami_env, prints_manager)
     prints_manager.execute_thread_prints(0)
 
+    prints_manager.add_print_job("Before load_conf_files", print_warning, 0)
+    prints_manager.execute_thread_prints(0)
```
Comment (yuvalbenshalom) [inline Tests/configure_and_test_integration_instances.py:659]: why did you add this?

File: Tests/configure_and_test_integration_instances.py [hunk 2]
```diff
@@ -230,6 +231,11 @@ def get_new_and_modified_integrations(git_sha1):
         isinstance(file_path, str) and checked_type(file_path, all_integration_regexes)
     ]
 
+    integrations_name_to_path_dict = dict([
+        (filepath_to_integration_name(file_path), file_path)
+        for file_path in (added_integration_files + modified_integration_files)
+    ])
+
```
Comment (yuvalbenshalom) [inline Tests/configure_and_test_integration_instances.py:238]: I am not sure if we need to create this data structure. lets check if we can check the path when we need it without crating this dict.

File: Tests/search_and_install_packs.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+    host = client.api_client.configuration.host
+    prints_manager.add_print_job('\nMaking "POST" request to server - "{}" to search pack {}.'.format(host, pack_id),
+                                 print, thread_index)
+
+    # make the search request
+    response_data, status_code, _ = demisto_client.generic_request_func(self=client,
+                                                                        path='/contentpacks/marketplace/search',
+                                                                        method='POST',
+                                                                        body={"packsQuery": pack_id, "size": 3},
```
Comment (yuvalbenshalom) [inline Tests/search_and_install_packs.py:46]: in order to test the search, we need to search the pack with its display name, not the id. thats because the user will see the display name and is expected to search that string, rather than the id.

BTW whats the meaning of the size here? why it is 3?

File: Tests/configure_and_test_integration_instances.py [hunk 4]
```diff
@@ -668,18 +680,39 @@ def main():
...
     if modified_integrations_names:
+
+        # Test packs search and installation - beginning of infrastructure
+        threads_list = []
+        packs_in_progress = set()  # used to avoid double installation
+        threads_prints_manager = ParallelPrintsManager(len(modified_integrations_names))
+        for index, int_name in enumerate(modified_integrations_names):
+            client = demisto_client.configure(base_url=servers[0], username=username, password=password,
+                                              verify_ssl=False)
+            thread = Thread(target=search_and_install_pack_and_its_dependencies,
```
Comment (yuvalbenshalom) [inline Tests/configure_and_test_integration_instances.py:707]: did you test this? I am not sure it can work now because our dependencies are not populated.

File: Tests/search_and_install_packs.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+            threads_list.append(t)
+
+        run_threads_list(threads_list)
+
+        pack_version = pack_metadata.get('currentVersion', '')
+        install_pack(client, prints_manager, thread_index, pack_id, pack_version)
+
+
+def search_and_install_pack_and_its_dependencies(client, prints_manager, thread_index, path, packs_in_progress):
```
Comment (yuvalbenshalom) [inline Tests/search_and_install_packs.py:152]: where do you handle the dependencies?

**Comment by dantavori** [comment]
Thanks build fails on Tanium Test, not related to the changes @yuvalbenshalom

File: Tests/Marketplace/Tests/search_and_install_packs_test.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    def __init__(self):
+        self.configuration = MockConfiguration()
+
+
+class MockClient:
+    def __init__(self):
+        self.api_client = MockApiClient()
+
+
+def test_search_and_install_packs_and_their_dependencies(mocker):
```
Comment (yuvalbenshalom) [inline Tests/Marketplace/Tests/search_and_install_packs_test.py:88]: please add documentation to every tests.
you can see the file`demisto_sdk/tests/integration_tests/secrets_integration_test.py` in the pr: https://github.com/demisto/demisto-sdk/pull/360/files

**Comment by yuvalbenshalom** [comment]
@dantavori please update from master
